services/data-info/src/data_info/routes/avus.clj

```diff
@@ -12,26 +13,29 @@
 
     (GET* "/avus" [:as {uri :uri}]
       :query [{:keys [user]} StandardUserQueryParams]
-      :return AVUGetResult
+      :return MetadataListing
+      :middlewares [wrap-metadata-base-url]
       :summary "List AVUs (administrative)"
-      :description (str "List iRODS AVUs associated with a data item. Include administrative/system AVUs."
-(get-error-code-block "ERR_NOT_A_USER, ERR_NOT_READABLE"))
+      :description
+          (str "List all AVUs associated with a data item. Include administrative/system iRODS AVUs."
+(get-error-code-block "ERR_NOT_A_USER, ERR_NOT_READABLE")
+(get-endpoint-delegate-block "metadata" "GET /avus/{target-type}/{target-id}"))
       (svc/trap uri meta/admin-metadata-get data-id))
 
-    (POST* "/avus" [:as {uri :uri}]
+    (PUT* "/avus" [:as {uri :uri}]
       :query [{:keys [user]} StandardUserQueryParams]
-      :body [{:keys [irods-avus]} (describe AVUListing "An list of AVUs to add")]
+      :body [{:keys [irods-avus]} (describe AVUListing "A list of AVUs to add")]
       :return AVUChangeResult
       :summary "Add AVUs (administrative)"
-      :description (str "Associate AVUs with a data item. Allow adding any AVU."
+      :description (str "Associate iRODS AVUs with a data item. Allow adding any AVU."
 (get-error-code-block "ERR_NOT_A_USER, ERR_NOT_READABLE, ERR_DOES_NOT_EXIST, ERR_NOT_WRITEABLE, ERR_NOT_AUTHORIZED"))
       (svc/trap uri meta/admin-metadata-add data-id irods-avus))
```

Comment:
re: the SO answer: this *is* a case with a specific ID, because it's working on the aggregate collection (whose ID is of the data-object, not the AVU). What they mean there is: if you don't have a specific ID (or want one generated, most often), then you need to use the nearest specific URI you have (usually the direct parent resource) but then PUT is inappropriate because you're not modifying *that* resource, you're creating a subresource instead, so use POST (especially since the RFC says "The POST method is used to request that the origin server accept the entity enclosed in the request as a new subordinate of the resource identified by the Request-URI in the Request-Line."). Their example makes this clear.

So, the PUT/POST distinction they're making would be between something like `POST /avus` to add an AVU with an auto-generated ID (non-specific, so use a POST to the parent resource) vs. `PUT /avus/:id` to add one with a specific AVU ID (specific, so use a PUT to the specific resource). So in fact what they're saying is exactly why I used POST for the add-avus endpoint: it's creating subresources.

It's weird because we're addressing a specific entity which is a collection of subresources, and the answer didn't specify that what they meant was "if you don't have a specific ID to PUT to, you have to use the specific ID of the parent resource and then POST is appropriate because you're creating a subresource". They didn't address the notion of modifying child resources by replacing the entirety of their parent, which is what our set endpoint does. But it's still replacing a resource with new contents, which is PUT.